cfme/tests/networks/test_sdn_crud.py

```diff
@@ -41,3 +43,26 @@ def test_sdn_crud(provider, appliance):
     provider.wait_for_delete()
 
     assert not network_provider.exists
+
+
+@pytest.mark.ignore_stream('5.9')
+@pytest.mark.provider([OpenStackProvider], scope='function', override=True)
+def test_router_crud(provider, appliance):
+    router_name = 'test_router_crud-{}'.format(fauxfactory.gen_alphanumeric().lower())
+    routers = appliance.collections.network_routers
+    router = routers.create(name=router_name,
+                            provider=provider,
+                            tenant=provider.data.networks.cloud_tenant,
+                            network_manager=provider.data.networks.network_manager)
+    assert router.exists
+
+    router.delete()
+    provider.refresh_provider_relationships()
+
+    def router_doesnt_exists():
+        # Navigate somewhere else then to Router Details to enforce page refresh.
+        collection = appliance.collections.network_providers
```

Comment:
@mshriver @quarckster Hi guys. The router.exists goes to router Details page to check whether the router exists, but once it is there, it doesn't try to refresh. Do you guys have some suggestion how can I do the refresh in some nicer way then navigating here and there?

Reply:
You can implement a `resetter()` method for the 'details' navigation destination that calls browser or UI toolbar refresh widgets.  This will be called after `prerequisite` and `step`.

Reply:
Note `resetter` is called even if navigation isn't needed (its on details page already, skips all navigation, but will call resetter.

Reply:
There are problems with the `not router.exists()` approach. It is simpler to just list all the routers and check whether the router is among them then to use negative of `router.exists`